demo/03-create-cluster.sh

```diff
@@ -99,6 +101,93 @@ IDENTITY_UAMIS_JSON_MAP=$(echo -n "${IDENTITY_UAMIS_JSON_MAP}" | jq \
 
 }
 
+get_identity_principal_id() {
+  local identity_name="$1"
+  local resource_group="$2"
+  
+  az identity show \
+    --name "${identity_name}" \
+    --resource-group "${resource_group}" \
+    --query principalId \
+    --output tsv
+}
+
+assign_role_to_identity() {
+  local principal_id="$1"
+  local role_name="$2"
+  local scope="$3"
+  
+  echo "Assigning role '${role_name}' to principal ID '${principal_id}' with scope '${scope}'"
+  
+  az role assignment create \
+    --assignee-object-id "${principal_id}" \
+    --assignee-principal-type ServicePrincipal \
+    --role "${role_name}" \
+    --scope "${scope}"
+    
+  echo "Role assignment completed successfully"
+}
+
+create_role_assignments_for_control_plane_identities() {
```

Comment:
This function and `create_role_assignments_for_data_plane_identities` are similar. 
Is it worth using parameters to generalize the logic for control-plane and data-plane identities?